### DISCUSSION TITLE
Workflow As a service

### DISCUSSION THREAD

**Discussion by mubukp** [elsa-workflows/elsa-core#2883] [Design & tech]
Hello,

We are trying to implement workflow as a service using Elsa 2.0 as the platform. In this regard request your inputs on below, 

1. How do we implement authentication/authorization for Elsa Host/Server APIs?
2. How to enable Service bus (may be MSMQ) to the Elsa APIs? So that even if Elsa is down, the requests from consumer applications are not lost and can be served as and when the Workflow host is up and running.
3. Multi tenancy
4. Any other considerations that would require to make Elsa run as "Workflow As a service" platform.

Thanks

**Comment by sfmskywalker** [comment]
Hi @mubukp 

1. Like with any other .NET project, you configure auth for Elsa server the same way.
2. [This page](https://elsa-workflows.github.io/elsa-core/docs/next/hosting/hosting-distributed-hosting) describes configuring Elsa to use a message broker.
3. Partially supported. I haven't used this myself, so I don't know how complete this is at the moment.
4. You probably want to host multiple nodes in a cluster to offer redundancy. Please see [this page](https://elsa-workflows.github.io/elsa-core/docs/next/hosting/hosting-distributed-hosting) for details on how to setup distributed hosting.
